packages/fields/types/Content/views/src/Controller.js

```diff
@@ -6,36 +6,26 @@ import memoizeOne from 'memoize-one';
 import TextController from '../../../Text/views/Controller/Controller';
 import * as paragraph from './editor/blocks/paragraph';
 
-const DEFAULT_BLOCKS = [paragraph];
-
-const flattenBlocks = (inputBlocks, outputObj = {}) =>
+const flattenBlocks = inputBlocks =>
   inputBlocks.reduce((outputBlocks, block) => {
+    // NOTE: It's enough to check just the type here as we've already flattened
+    // and deduped dependencies during build.
     if (outputBlocks[block.type]) {
-      // check the referential equality of a blocks Node since it has to be
-      // defined and if they're equal we know it's the same block
-      if (outputBlocks[block.type].Node !== block.Node) {
-        throw new Error(`There are two different Content blocks with the type '${block.type}'`);
-      } else {
-        // This block (and its dependencies) have already been added
-        return outputBlocks;
-      }
+      throw new Error(
+        `Encountered more than one Content block with type of '${
+          block.type
+        }'. Content blocks must have globally unique types.`
+      );
     }
 
     if (block.Node === undefined) {
-      throw new Error(`Unable to load Content block '${block.type}' - no Node component defined`);
+      throw new Error(`Unable to load Content block '${block.type}': no 'Node' export found.`);
     }
 
-    const { dependencies, ...blockToInsert } = block;
-
-    outputBlocks[block.type] = blockToInsert;
-
-    if (dependencies !== undefined) {
-      // Recurse on the dependencies
-      flattenBlocks(dependencies, outputBlocks);
```

Comment:
Dependencies are already flattened pre-build in `Implemenation.js`, so there's nothing to do here.